### DISCUSSION TITLE
Fix packages versions

### DISCUSSION THREAD

**Discussion by bouda1** [microsoft/vcpkg#40651] [General]
Hi,

Maybe this question has already been posted and maybe there are already answers. I'm pretty sure someone will be able to show me the way. We have looked for a solution but did not succeed until now.

We work with a vcpkg.json file in our project, here is the beginning of it:
```
{
  "dependencies": [
    {
      "name": "abseil",
      "features": [
        "cxx17"
      ]
    },
    "fmt",
    "grpc",
    "spdlog",
    "boost-asio",
    "boost-multi-index",
    "boost-process",
    "boost-program-options",
    "rapidjson",
    "gtest",
    {
      "name": "libssh2",
      "platform": "linux"
    },
    {
      "name": "curl",
      "platform": "linux"
    },
```

At the first compilation, packages are downloaded and compiled and then we use them as they are.

Sometimes, one of them has its version updated but until we force an update on our side we are not aware of that.

But then I new collaborator joins the team, he tries to compile the project and cries because it doesn't compile anymore ; in fact he's just downloaded the last versions available.

So here is my question, how can I force a given version for each of my dependencies, for example libssh2 1.9.7  ?

Thank you very much, cheers.
David.

**Comment by dg0yt** [comment]
You could force individual versions with `overrides`.
But you probably want to start with defining the `builtin-baseline`.
Both here:
https://learn.microsoft.com/en-us/vcpkg/users/versioning#version-constraints

**Reply by Osyotr** [reply]
Adding to the "use baselines" advise, I suggest:
1) Use `--x-abi-tools-use-exact-versions`
2) Use the same version of vcpkg-scripts on every machine (e.g. vcpkg repo is checked out at the same commit). I use this script for that purpose https://gist.github.com/Osyotr/e2d415312a10183bc7c614013494bf21

**Reply by dg0yt** [reply]
FTR the advice by @Osyotr is for sharing binary artifact over different machines. It is not related to the problem of fixing versions, but relevant for fixing binary artifacts for the same version (given a shared cache).

**Comment by bouda1** [comment]
Thank you for your answers, it works as expected.
